pkg/util/files/manager.go

```diff
@@ -16,35 +16,38 @@ package files
 
 import (
 	"log"
+	"os"
 	"path/filepath"
 	"regexp"
 
 	"github.com/spf13/afero"
 )
 
-//go:generate mockgen -source=creator.go -destination=creator_mock.go -package=files FileCreator
+//go:generate mockgen -source=manager.go -destination=manager_mock.go -package=files FileManager
 
-//FileCreator is an interface to encapsulate the file creation process. Is has 2 implementations
-//in memory and one that uses files on disk.
-type FileCreator interface {
+//FileManager is an interface to encapsulate the file read/creation process. It has 2 implementations
+type FileManager interface {
```

Comment:
@patrickpichler , @didiladi so you'd argue for dropping this in favour of using afero directly?

I generally agree, there is no reason to wrap an abstraction of the filesystem in another abstraction - I'm fine if we decide to go all afero for that and drop our own abstraction. 

However the methods in here do additional things as far as I can tell. 
They sanitize inputs, join filepaths, react to errors and return different outputs. 

Would that mean that you want to move the input validation/preparation and error handling into the calling code? 

If we need to do these things, I'd stick with these methods as they seem to be shared utilities more so than an abstraction layer. 
We could of course keep the methods and have callers pass in an `afero.Fs` - so that we could use afero methods directly where needed, instead of hiding it behind our FileManager - but retain the shared steps.

Reply:
Most of the methods are anyway thin wrappers around afero and do not bring any 
real benefit. I would drop all of them.

Reply:
I'm closing this pull request and opening a new one with the Afero only implementation